tachyon/crypto/sumcheck/multilinear/sumcheck_verifier.h

```diff
@@ -187,31 +187,42 @@ F InterpolateUniPoly(const std::vector<F>& poly, const F& evaluation_point) {
   size_t chunk_size = (poly_size + thread_nums - 1) / thread_nums;
   size_t num_chunks = (poly_size + chunk_size - 1) / chunk_size;
 
-  std::vector<F> products(num_chunks, F::One());
-  std::vector<F> denom_ups(num_chunks, F::One());
-  std::vector<std::vector<F>> list_of_evals(num_chunks);
-  OMP_PARALLEL_FOR(size_t i = 0; i < num_chunks; ++i) {
-    size_t begin = i * chunk_size;
-    size_t len = (i == num_chunks - 1) ? poly_size - begin : chunk_size;
-    list_of_evals[i].reserve(len);
-    F check = F(begin);
-    for (size_t j = begin; j < begin + len; ++j) {
-      const F difference = evaluation_point - check;
-      list_of_evals[i].push_back(difference);
-      products[i] *= difference;
-      if (j > 1) {
-        denom_ups[i] *= check;
+  F product;
+  F denom_up;
+  std::vector<F> evals;
+  {
+    std::vector<F> products(num_chunks, F::One());
+    std::vector<F> denom_ups(num_chunks, F::One());
+    std::vector<std::vector<F>> list_of_evals(num_chunks);
+    OMP_PARALLEL_FOR(size_t i = 0; i < num_chunks; ++i) {
+      size_t begin = i * chunk_size;
+      size_t len = (i == num_chunks - 1) ? poly_size - begin : chunk_size;
+      list_of_evals[i].reserve(len);
+      F check = F(begin);
+      for (size_t j = begin; j < begin + len; ++j) {
```

Comment:
```suggestion
      for (size_t j = 0; j < len; ++j) {
```
Shouldn't it be like this since below if statement is j > 1?

Reply:
Since `denom_up` is trying to compute (|poly_size| - 1)!, so this is right.